### PR TITLE
[ENG-2898] Collections submission failure

### PR DESCRIPTION
**Pull request by futa-ikeda**
- Ticket: [ENG-2898]
- Feature flag: n/a

## Purpose
- Fix bug discovered in Ember upgrade testing where submission to a collection fails
- Fix another bug where the last panel of the submission page (`Collection Metadata`) does not load properly

## Summary of Changes
- Update places where the CollectedMetadatum's Collection relationship is not properly loaded

## Side Effects
- None
## QA Notes
- Submissions should now be able to be submitted, and no errors should be in the browser console
- Sometimes the `Collection Metadata` panel would not load properly if you are on the page `http://osf.io/collections/mycollection/submit` and reload the page, and this issue has been addressed in this PR as well

[ENG-2898]: https://openscience.atlassian.net/browse/ENG-2898

### CODE REVIEW

File: app/models/collected-metadatum.ts [hunk 1]
```diff
@@ -53,10 +53,13 @@ export default class CollectedMetadatumModel extends OsfModel.extend(Validations
 
     @computed('collection.displayChoicesFields.[]')
     get displayChoiceFields() {
-        return choiceFields
-            .filter(field => this.collection
-                .get('displayChoicesFields')
-                .includes(`${field}Choices` as ChoicesFields));
+        if (this.collection.get('displayChoicesFields')) {
```
Comment (fabmiz) [inline app/models/collected-metadatum.ts:56]: can we do without the `.get` here?

File: app/models/collected-metadatum.ts [hunk 2]
```diff
@@ -53,10 +53,13 @@ export default class CollectedMetadatumModel extends OsfModel.extend(Validations
...
+            return choiceFields
+                .filter(field => this.collection
+                    .get('displayChoicesFields')
```
Comment (fabmiz) [inline app/models/collected-metadatum.ts:59]: Ditto 👆🏽

File: app/models/collected-metadatum.ts (hunk 1 above)
Comment (futa-ikeda) [reply app/models/collected-metadatum.ts:56]: In this case, `collection` is still a proxy, so we'll have to use the `.get` here.

Comment (fabmiz) [reply app/models/collected-metadatum.ts:56]: ah proxies 😩 Depending on a async relationship is best practices but that can be addressed later.

**Review by fabmiz — DISMISSED** [review]
🥇

File: lib/collections/addon/components/collections-submission/component.ts [hunk 3]
```diff
@@ -89,6 +89,7 @@ export default class Submit extends Component {
         }
 
         this.collectedMetadatum.set('guid', this.collectionItem);
+        this.collectedMetadatum.set('collection', this.collection);
```
Comment (adlius) [inline lib/collections/addon/components/collections-submission/component.ts:92]: I don't think you need to set the `collection` here again.

**Review by adlius — APPROVED** [review]

**Pull request merged**